lambdas/choice/src/writer.py

```diff
@@ -1,6 +1,89 @@
 import os
+import boto3
+
+"""
+Processes SNS messages containing user choices by writing choices to Dynamo.
+For event and database record shapes, see https://github.com/sjawhar/enchanted-brain/issues/4.
+"""
+
+
+EVENT_USER_ID_KEY = 'userId'
+EVENT_CHOICE_TYPE_KEY = 'choiceType'
+EVENT_CHOICE_KEY = 'choice'
+EVENT_TIMESTAMP_KEY = 'timestamp'
+
+CHOICE_COLOR = 'CHOICE_COLOR'
+CHOICE_EMOTION = 'CHOICE_EMOTION'
+CHOICE_CHILLS = 'CHOICE_CHILLS'
+CHOICE_IMAGERY = 'CHOICE_IMAGERY'
+
+CONCERT_SONG_ID = os.environ['EVENT_NAME']
+DYNAMO_DB_TABLE_NAME = os.environ['DYNAMO_TABLE_NAME']
+
+choice_type_to_choice_key = {
+    CHOICE_COLOR: 'colors',
+    CHOICE_EMOTION: 'emotions',
+    CHOICE_CHILLS: 'chills',
+    CHOICE_IMAGERY: 'imagery'
+}
+choice_types = choice_type_to_choice_key.keys()
+
+dynamodb = boto3.resource(
+    'dynamodb',
+    region_name=os.environ['AWS_DYNAMO_REGION'],
+    endpoint_url=os.environ['AWS_DYNAMO_ENDPOINT'],
+    aws_secret_access_key=os.environ['AWS_SECRET_ACCESS_KEY'],
+    aws_access_key_id=os.environ['AWS_ACCESS_KEY_ID']
+)
+table = dynamodb.Table(DYNAMO_DB_TABLE_NAME)
+
 
 def handler(event, context):
-  print(event)
-  print(os.environ)
-  return { 'statusCode': 204 }
+    return write_to_dynamo(dynamodb, event)
+
+
+def write_to_dynamo(table, event):
+
+    # Get data from event
+    listen_id = event[EVENT_USER_ID_KEY]
+    choice_type = event[EVENT_CHOICE_TYPE_KEY]
+    choice = event[EVENT_CHOICE_KEY]
+    timestamp = event[EVENT_TIMESTAMP_KEY]
+
+    # If choice type not recognized, return 422
+    if choice_type not in choice_types:
+        return {'statusCode': 422}
```

Comment:
I've only works with Lambda functions in Node, so please correct me if it's different in Python.

This function is being invoked by SNS, so statusCode is not meaningful. In fact, returning anything indicates to SNS that the function completed successfully. Instead, what you want to do is raise an error. This will cause a retry and, after 3 failures, redirect the message to a dead letter queue (if configured).

Reply:
Nope, you're right. Fixed.